### PR TITLE
RadialIndicator visible to observer/allies & Different PowerDelta label when power sabotaged

### PR DESCRIPTION
**Pull request by chaserli**
In vanilla game, when the player selects a defense building, the radial indicator is shown only if it belongs to the player. Now the indicator is also drawn for observers. For other houses it can be specified by

In `rulesmd.ini`
```ini
[AudioVisual]
RadialIndicatorVisibility=allies  ; list of Affected House Enumeration (allies/ally | enemies/enemy | all)
```

When your power is sabotaged by a spy, the number given by the power delta counter may look confusing to players. Now I replaced it by ⚡❌, which can also be replaced by

In `uimd.ini`
```ini
[ToolTips]
PowerBlackoutLabel=  ; CSF entry key
```
If `[GlobalControls]->DebugKeysEnabled=yes` power delta is still displayed

### CODE REVIEW

**Comment by Starkku** [comment]
Code looks alright to me. The visibility control could also be implemented at per-building level with it defaulting to the global setting but I am unsure if there would be need for that degree of customization.

**Review by Starkku — APPROVED** [review]
Code looks alright to me. The visibility control could also be implemented at per-building level with it defaulting to the global setting but I am unsure if there would be need for that degree of customization.

**Comment by FS-21** [comment]
> Code looks alright to me. The visibility control could also be implemented at per-building level with it defaulting to the global setting but I am unsure if there would be need for that degree of customization.

The per-building customization can be used for example for showing the sensors range in defenses capable to detect stealth diguised units.

File: src/Phobos.cpp [hunk 1]
```diff
@@ -227,6 +228,9 @@ DEFINE_HOOK(0x5FACDF, OptionsClass_LoadSettings_LoadPhobosSettings, 0x5)
 		pINI_UIMD->ReadString(TOOLTIPS_SECTION, "PowerLabel", NONE_STR, Phobos::readBuffer);
 		Phobos::UI::PowerLabel = GeneralUtils::LoadStringOrDefault(Phobos::readBuffer, L"\u26a1"); // ⚡
 
+		pINI_UIMD->ReadString(TOOLTIPS_SECTION, "PowerBlackoutLabel", NONE_STR, Phobos::readBuffer);
+		Phobos::UI::PowerBlackoutLabel = GeneralUtils::LoadStringOrDefault(Phobos::readBuffer, L"\u26a1\u2704"); // ⚡✄
```
Comment (chaserli) [inline src/Phobos.cpp:232]: I need a better default symbol for this one. My hand drawing is poor so I cannot add this one to the font yet

File: src/Misc/Hooks.UI.cpp [hunk 2]
```diff
@@ -198,3 +201,16 @@ DEFINE_HOOK(0x6D4684, TacticalClass_Draw_FlyingStrings, 0x6)
...
 }
+
+DEFINE_HOOK(0x456776, BuildingClass_DrawRadialIndicator_Visibility, 0x6)
+{
+	enum { ContinueDraw = 0x456789, DoNotDraw = 0x456962 };
+	GET(BuildingClass* const, pThis, ESI);
+
+	if (HouseClass::IsCurrentPlayerObserver()
+		|| pThis->Owner->IsControlledByCurrentPlayer()
+		|| EnumFunctions::CanTargetHouse(RulesExt::Global()->RadialIndicatorVisibility, pThis->Owner, HouseClass::CurrentPlayer))
```
Comment (chaserli) [inline src/Misc/Hooks.UI.cpp:212]: I don't think per BuildingType config is necessary.  You can differentiate with `ConcentricRadialIndicator` for example

File: src/Phobos.cpp (hunk 1 above)
Comment (chaserli) [reply src/Phobos.cpp:232]: So how about disabling power delta display during power blackout?

Comment (Starkku) [reply src/Phobos.cpp:232]: Don't know about hiding the power delta display. Color change would probably be better, defaults to something like gray.

**Pull request merged**